xchange-core/src/main/java/org/knowm/xchange/service/trade/params/TradeHistoryParamsAll.java

```diff
@@ -30,6 +34,8 @@
   private Long offset;
   private CurrencyPair pair;
   private Collection<CurrencyPair> pairs = Collections.emptySet();
+  private Instrument instrument;
+  private Collection<Instrument> instruments = Collections.emptySet();
```

Comment:
this should replace the `pair` / `pairs` fields rather then co-oexist with them

you can implement the *CurrencyPair* method using just `instument` / `instruments`

Reply:
Implemented in #cbfcdf7